core/client/src/transaction_builder/transaction_builder.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! The runtime api for building transactions.
+
+use runtime_primitives::AnySignature;
+use sr_api_macros::decl_runtime_apis;
+
+decl_runtime_apis! {
+	/// The `TransactionBuilder` trait that provides required functions
+	/// for building a transaction from a Call for the runtime.
+	pub trait TransactionBuilder {
+		/// Construct the payload.
```

Comment:
Could improve the docs a bit more? How is this supposed to be used?

Reply:
Good question, honestly I don't have it clear. I updated the description according to the way I think it should be used but I feel we may be missing parameters in the functions.